### PR TITLE
Sas drag bloke implementation

### PR DESCRIPTION
**Pull request by Kubsy**
Sas drag bloke implemented.

### CODE REVIEW

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		-BLOCK(1.0f / 2), -460
+	),
+		std::pair(
+			EulerAngles(ANGLE(-10.0f), ANGLE(-30.0f), 0),
+			EulerAngles(ANGLE(10.0f), ANGLE(30.0f), 0)
+		)
+};
+
+
+void InitialiseDragSAS(short itemNumber)
```
[the remark this answers is not in the document]
Comment (Kubsy) [reply TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp]: done, also removed for sarcophagus

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+{
+
+}
+
+void DragSASCollision(short itemNumber, ItemInfo* laraItem, CollisionInfo* coll)
+{
+	auto* item = &g_Level.Items[itemNumber];
+	auto* laraInfo = GetLaraInfo(laraItem);
+
+	long x, z;
```
[the remark this answers is not in the document]
Comment (Kubsy) [reply TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp]: done

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		std::pair(
+			EulerAngles(ANGLE(-10.0f), ANGLE(-30.0f), 0),
+			EulerAngles(ANGLE(10.0f), ANGLE(30.0f), 0)
+		)
+};
+
+void DragSASCollision(short itemNumber, ItemInfo* laraItem, CollisionInfo* coll)
+{
+	auto* item = &g_Level.Items[itemNumber];
+	auto* laraInfo = GetLaraInfo(laraItem);
```
Comment (Sezzary) [inline TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp:34]: `laraInfo` can be simply `lara` like everywhere else.

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		laraInfo->Control.HandStatus == HandStatus::Free &&
+		!(laraItem->Animation.IsAirborne) &&
+		!(item->Flags & IFLAG_ACTIVATION_MASK) ||
+		laraInfo->Control.IsMoving && laraInfo->InteractedItem == itemNumber)
+	{
+		if (TestLaraPosition(DragSasBounds, item, laraItem))
+		{
+			if (MoveLaraPosition(DragSASPosition, item, laraItem))
+			{
+
```
Comment (Sezzary) [inline TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp:48]: No need for a newline here. :)

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
+#include "framework.h"
```
Comment (Sezzary) [inline TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp:1]: Includes should be organised like this:
```
#include "framework.h"
#include "tr4_sas_drag_bloke.h"

#include "Game/collision/collide_item.h"
#include "Game/health.h"
#include "Game/items.h"
#include "Game/Lara/lara.h"
#include "Game/Lara/lara_helpers.h"
#include "Game/pickup/pickup.h"
#include "Math/Math.h"
#include "Specific/Input/Input.h"
#include "Specific/level.h"
#include "Specific/setup.h"
```

File: TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		-100, 100, 
+		-BLOCK(1.0f / 2), -460
+	),
+		std::pair(
+			EulerAngles(ANGLE(-10.0f), ANGLE(-30.0f), 0),
+			EulerAngles(ANGLE(10.0f), ANGLE(30.0f), 0)
+		)
+};
+
+void DragSASCollision(short itemNumber, ItemInfo* laraItem, CollisionInfo* coll)
```
Comment (Sezzary) [inline TombEngine/Objects/TR4/Object/tr4_sas_drag_bloke.cpp:31]: Three-letter acronyms such as "SAS" may be treated like words and formatted as "Sas" everywhere.

**Pull request merged**